src/ocean/task/TaskPool.d

```diff
@@ -153,6 +153,36 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
         return true;
     }
 
+    /***************************************************************************
+
+        Suspends the current task until all tasks in the pool have finished
+        executing.
+
+    ***************************************************************************/
+
+    public void awaitAllTasks ()
+    {
+        scope iter = this.new ReadOnlyBusyItemsIterator;
```

Comment:
My memory is vague here - does `ReadOnly` here implies you shouldn't modify task items? (because you do)